server/kafka/consumer.go

```diff
@@ -261,3 +309,68 @@ func (c *saramaConsumer) ConsumeClaim(sess sarama.ConsumerGroupSession, claim sa
 
 	return nil
 }
+
+// Retrieve the schema of the message and deserialize it.
+func (c *saramaConsumer) deserializePayload(payload []byte) ([]byte, error) {
+	// first byte of the payload is 0
+	if payload[0] != byte(0) {
+		return nil, fmt.Errorf("magic byte is not 0")
```

Comment:
Can we change this to something like "failed to deserialize payload: magic byte is not 0"?

Reply:
Done.